core-rust/core-api-server/src/core_api/conversions/receipt.rs

```diff
@@ -402,13 +307,26 @@ pub fn to_api_state_updates(
     let mut created_substates = Vec::new();
     let mut updated_substates = Vec::new();
     let mut deleted_substates = Vec::new();
+
+    // Step 1 - First, build actions
+    let mut changes_to_map = Vec::new();
```

Comment:
It's actually rather gnarly because `create_typed_substate_key` returns a result. And propogating that before doing the filter is nasty. I kinda want a `filter_map_with_error` which can work on a `Result<Option<X>, T>`

You can achieve this with separating the `map` and the `collect` and using a gnarly `match` statement, but I think it all becomes less readable at this point. So if you're happy, I'll leave as is :)